src/fty_common_socket_basic_mailbox_server.cc

```diff
@@ -0,0 +1,379 @@
+/*  =========================================================================
+    fty_common_socket_basic_mailbox_server - Basic synchronous mailbox server using unix socket
+
+    Copyright (C) 2014 - 2019 Eaton
+
+    This program is free software; you can redistribute it and/or modify
+    it under the terms of the GNU Lesser General Public License as published by
+    the Free Software Foundation; either version 2 of the License, or
+    (at your option) any later version.
+
+    This program is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU General Public License for more details.
+
+    You should have received a copy of the GNU Lesser General Public License along
+    with this program; if not, write to the Free Software Foundation, Inc.,
+    51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+    =========================================================================
+*/
+
+/*
+@header
+    fty_common_socket_basic_mailbox_server - Basic synchronous mailbox server using unix socket
+@discuss
+@end
+*/
+#include "fty_common_socket_basic_mailbox_server.h"
+
+#include <errno.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <sys/types.h>
+#include <sys/socket.h>
+#include <pwd.h>
+#include <sys/un.h>
+#include <unistd.h>
+#include <stdexcept>
+
+#include "fty_common_socket_helpers.h"
+
+//  Structure of our class
+namespace fty
+{
+
+    SocketBasicServer::SocketBasicServer(   fty::SyncServer & server,
+                                            const std::string & path,
+                                            size_t maxClient)
+     : m_server(server), m_path(path), m_maxClient(maxClient)
+    {
+        m_serverSocket = -1;
+        init();
+    }
+    
+    void SocketBasicServer::init()
+    {
+        if(m_serverSocket != -1)
+        {
+            throw std::runtime_error("already initialize");
+        }
+        
+        struct sockaddr_un name;
+        int ret;
+
+        /*
+        * In case the program exited inadvertently on the last run,
+        * remove the socket.
+        */
+
+        unlink(m_path.c_str());
+
+        /* Create local socket. */
+
+        m_serverSocket = socket(AF_UNIX, SOCK_STREAM, 0);
```

Comment:
PF_UNSPEC (0)

Reply:
Why it should be PF_UNSPEC  instead of AF_UNIX (or PF_UNIX) when we want to use unix socket ?

Reply:
* The comment line 73 is not really relevant and will be change to "Create Unix socket"

Reply:
//socket.h
#define PF_UNSPEC	AF_UNSPEC  (where AF_UNSPEC is defined as 0, PF_* == AF_*)
Just a way to use a named define instead of hardcoding a constant value (3rd arg of connect())
Very cosmetic, ignore the comment...

Reply:
I just didn't realize what was the 3rd arguments.